homeassistant/components/knx/config_flow.py

```diff
@@ -100,6 +105,11 @@ async def async_step_connection_type(
     ) -> FlowResult:
         """Handle connection type configuration."""
         if user_input is not None:
+            self._async_scan_gen = None  # stop the scan
```

Comment:
Garbage collect can be delayed for a long time. Seams like a bad way to stop the background scanner. I think you should explicitly call aclose() on the iterator.

Reply:
It exits the async generator after the timeout of 2 seconds in any way.

Reply:
Maybe. But it would be a hidden change if changed in the lib later, so please close it when done.